Reinforced cameras / ai cameras and router

snowywinters opened the pull request:
## About the PR
Adds reinforced cameras and a new sebnet for Ai cameras/routers

## Why / Balance
https://discord.com/channels/968983104247185448/1447879561122418731
main reason was because meteors could take out the cameras from over 5 tiles away and the normal camera was too easy to take out by random stuff

For the ai camera subnet is is so the Ai core cameras don't take up space in commands subnet and being its own area it should have its own 

## Technical details
the reinforced cameras have the following
1. health 275 
2. explosion resistance 75% 
NOTE: for meteors and any c4 or bombs like it can take out the cameras 1 tile away and can only survive one non direct impact from a meteor and a direct impact will take it out
3. can be repaired with a welder
4. has the Electronic damage modifier set which has 30% blunt, 50% slash, 30% Piercing, shock 2, heat 3, and Structural 50%
5. has the security wire panel so you can add steel/plasteel to the wire panel it only needs one steel/plasteel instead of two
6. has the full deconstruction route that works and gives mats back
7. for construction look below in media
8. command and ai mapping cameras have plasteel the rest have steel in the wire panel
i don't believe i forgot anything 

AI camera subnet
1. adds a AI subnet
2. adds a mapping camera for the Ai subnet
3. adds a camera router for the Ai subnet


## Media
<img width="335" height="266" alt="Screenshot 2025-12-23 231815" src="https://github.com/user-attachments/assets/5e95911f-63e6-48aa-8315-fc2621333338" />

<img width="71" height="109" alt="Screenshot 2025-12-23 231927" src="https://github.com/user-attachments/assets/af263283-7509-405f-98e1-5ff41aa52774" />


## Requirements
<!-- Confirm the following by placing an X in the brackets: [X] -->
- [x] I have tested all added content and changes.
- [x] I have added media to this PR or it does not require an ingame showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes


**Changelog**

:cl:
- add: Reinforced cameras can now be built! They have additional HP and are more resistant to explosions and meteors.
- add: The AI has its own section in the security cameras console.

snowywinters committed:
AI cam router

snowywinters committed:
reinforced camera try one

snowywinters committed:
reinforced camera try two

snowywinters committed:
reinforced camera try 3

snowywinters committed:
reinforced camera try 4

snowywinters committed:
reinforced camera try 5

snowywinters committed:
reinforced camera try 6

snowywinters committed:
reinforced camera try 7

snowywinters committed:
reinforced camera try 8

snowywinters committed:
AI router fix

snowywinters committed:
test balance

snowywinters committed:
Test construction rework

snowywinters committed:
name fix

snowywinters committed:
Merge branch 'master' into reinforced-cams

Signed-off-by: snowywinters <134970424+snowywinters@users.noreply.github.com>

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

ShepardToTheStars commented on Resources/Prototypes/_DV/Entities/Structures/Wallmounts/WallmountMachines/reinforced_surveillance_camera.yml:
```diff
@@ -0,0 +1,622 @@
+    receiveFrequencyId: SurveillanceCameraSecurity
+    transmitFrequencyId: SurveillanceCamera
+  - type: SurveillanceCamera
+    networkSet: true
```
```suggestion
  parent: SurveillanceCameraSecurity
  id: SurveillanceCameraSecurityReinforced
  name: Reinforced camera
  suffix: Security
  components:
```

ShepardToTheStars commented on Resources/Prototypes/_DV/Entities/Structures/Wallmounts/WallmountMachines/reinforced_surveillance_camera.yml:
```diff
@@ -0,0 +1,622 @@
+    receiveFrequencyId: SurveillanceCameraScience
+    transmitFrequencyId: SurveillanceCamera
+  - type: SurveillanceCamera
+    networkSet: true
```
```suggestion
  parent: SurveillanceCameraScience
  id: SurveillanceCameraScienceReinforced
  name: reinforced camera
  suffix: Epistemics
  components:
```

ShepardToTheStars commented on Resources/Prototypes/_DV/tags.yml:
```diff
@@ -222,6 +222,37 @@
+
+- type: Tag
+  id: cameraai
+#reinforced cams end
```
Reason for the tags? I know they look to be used during the construction graph stuff but normal cameras don't use tags so I'm wondering if they are even needed in the first place?

ShepardToTheStars commented on Resources/Prototypes/_DV/Entities/Structures/Wallmounts/WallmountMachines/reinforced_surveillance_camera.yml:
```diff
@@ -0,0 +1,622 @@
+    receiveFrequencyId: SurveillanceCameraSupply
+    transmitFrequencyId: SurveillanceCamera
+  - type: SurveillanceCamera
+    networkSet: true
```
```suggestion
  parent: SurveillanceCameraSupply
  id: SurveillanceCameraSupplyReinforced
  name: reinforced camera
  suffix: Supply
  components:
```

ShepardToTheStars commented on Resources/Prototypes/_DV/Entities/Structures/Wallmounts/WallmountMachines/reinforced_surveillance_camera.yml:
```diff
@@ -0,0 +1,622 @@
+      damageCoefficient: 0.25
+    - type: Tag
+      tags:
+        - cameracon
```
```suggestion
- type: entity
  parent: SurveillanceCameraReinforcedBase
  id: SurveillanceCameraReinforcedConstructed
  suffix: Constructed
  components:
    - type: DeviceNetwork
      deviceNetId: Wired
      transmitFrequencyId: SurveillanceCamera
    - type: SurveillanceCamera
      setupAvailableNetworks:
        - SurveillanceCameraEngineering
        - SurveillanceCameraSecurity
        - SurveillanceCameraService
        - SurveillanceCameraSupply
        - SurveillanceCameraScience
        - SurveillanceCameraGeneral
        - SurveillanceCameraMedical
        - SurveillanceCameraCommand
        - SurveillanceCameraAI
    - type: Construction
      graph: ReinforcedSurveillanceCameraGraph
      node: camera
    - type: Tag
      tags:
        - cameracon
```

snowywinters commented on Resources/Prototypes/_DV/Entities/Structures/Wallmounts/WallmountMachines/reinforced_surveillance_camera.yml:
```diff
@@ -0,0 +1,622 @@
+    damageCoefficient: 0.25
+  - type: Tag
+    tags:
+      - cameraeng
```
I can try to do this suggestion when i get back home to see if it works or not if it does work right it would clear a lot of the code up

snowywinters committed:
New reinforced camera code

Recommend by ShepardToTheStars thanks a lot for the recommendation and code start

snowywinters committed:
fixed the one missing lv wire

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

Cepelinas1 commented:
approved

snowywinters committed:
Merge remote-tracking branch 'upstream/master' into reinforced-cams

snowywinters committed:
Name fix agian

snowywinters committed:
removed the extra frequency

snowywinters committed:
AHHHHHHHHH

snowywinters committed:
fuck

Toby222 commented on Resources/Textures/_DV/Structures/Wallmounts/Reinforced_camera.rsi/meta.json:
```diff
@@ -0,0 +1,61 @@
+        "y": 32
+    },
+    "license": "CC-BY-SA-3.0",
+    "copyright": "Taken from /tg/station at https://github.com/tgstation/tgstation/commit/6bfe3b2e4fcbcdac9159dc4f0327a82ddf05ba7bi | Reinforced camera from snowy_winter",
```
Typo in url
```suggestion
    "copyright": "Taken from /tg/station at https://github.com/tgstation/tgstation/commit/6bfe3b2e4fcbcdac9159dc4f0327a82ddf05ba7b | Reinforced camera from snowy_winter",
```

ShepardToTheStars commented on Resources/Prototypes/_DV/Entities/Structures/Wallmounts/WallmountMachines/reinforced_surveillance_camera.yml:
```diff
@@ -0,0 +1,279 @@
+  components:
+  - type: Sprite
+    drawdepth: WallMountedItems
+    sprite: _DV\Structures\Wallmounts\Reinforced_camera.rsi
```
Debug assertion error on start-up.
```suggestion
    sprite: _DV/Structures/Wallmounts/Reinforced_camera.rsi
```

ShepardToTheStars commented on Resources/Prototypes/_DV/Recipes/Construction/Graphs/utilities/reinforced_cameras.yml:
```diff
@@ -0,0 +1,708 @@
+      conditions:
+      - !type:WirePanel {}
+      - !type:HasTag
+        tag: cameraai
```
```suggestion
        tag: ReinforcedCameraAI
```

ShepardToTheStars commented on Resources/Prototypes/_DV/Recipes/Construction/Graphs/utilities/reinforced_cameras.yml:
```diff
@@ -0,0 +1,708 @@
+      conditions:
+      - !type:WirePanel {}
+      - !type:HasTag
+        tag: cameracon
```
```suggestion
        tag: ReinforcedCameraConstructed
```

ShepardToTheStars commented on Resources/Prototypes/_DV/Recipes/Construction/Graphs/utilities/reinforced_cameras.yml:
```diff
@@ -0,0 +1,708 @@
+      conditions:
+      - !type:WirePanel {}
+      - !type:HasTag
+        tag: cameracon
```
```suggestion
        tag: ReinforcedCameraConstructed
```

ShepardToTheStars commented on Resources/Prototypes/_DV/Entities/Structures/Wallmounts/WallmountMachines/reinforced_surveillance_camera.yml:
```diff
@@ -0,0 +1,279 @@
+    securityLevel: medSecurity
+  - type: Tag
+    tags:
+      - cameragen
```
```suggestion
      - ReinforcedCameraGeneral
```

ShepardToTheStars commented on Resources/Prototypes/_DV/Recipes/Construction/Graphs/utilities/reinforced_cameras.yml:
```diff
@@ -0,0 +1,708 @@
+      conditions:
+      - !type:WirePanel {}
+      - !type:HasTag
+        tag: camerasec
```
```suggestion
        tag: ReinforcedCameraSecurity
```

ShepardToTheStars commented on Resources/Prototypes/_DV/Entities/Structures/Wallmounts/WallmountMachines/reinforced_surveillance_camera.yml:
```diff
@@ -0,0 +1,279 @@
+    securityLevel: medSecurity
+  - type: Tag
+    tags:
+      - camerasci
```
```suggestion
      - ReinforcedCameraScience
```

ShepardToTheStars commented on Resources/Prototypes/_DV/Entities/Structures/Wallmounts/WallmountMachines/reinforced_surveillance_camera.yml:
```diff
@@ -0,0 +1,279 @@
+    securityLevel: medSecurity
+  - type: Tag
+    tags:
+      - camerasup
```
```suggestion
      - ReinforcedCameraSupply
```

ShepardToTheStars commented on Resources/Prototypes/_DV/Recipes/Construction/Graphs/utilities/reinforced_cameras.yml:
```diff
@@ -0,0 +1,708 @@
+      conditions:
+      - !type:WirePanel {}
+      - !type:HasTag
+        tag: camerasup
```
```suggestion
        tag: ReinforcedCameraSupply
```

ShepardToTheStars commented on Resources/Prototypes/_DV/Recipes/Construction/Graphs/utilities/reinforced_cameras.yml:
```diff
@@ -0,0 +1,708 @@
+      conditions:
+      - !type:WirePanel {}
+      - !type:HasTag
+        tag: camerasup
```
```suggestion
        tag: ReinforcedCameraSupply
```

ShepardToTheStars commented on Resources/Prototypes/_DV/Entities/Structures/Wallmounts/WallmountMachines/reinforced_surveillance_camera.yml:
```diff
@@ -0,0 +1,279 @@
+    securityLevel: medSecurity
+  - type: Tag
+    tags:
+      - cameraser
```
```suggestion
      - ReinforcedCameraService
```

ShepardToTheStars commented on Resources/Prototypes/_DV/Recipes/Construction/Graphs/utilities/reinforced_cameras.yml:
```diff
@@ -0,0 +1,708 @@
+      conditions:
+      - !type:WirePanel {}
+      - !type:HasTag
+        tag: cameraser
```
```suggestion
        tag: ReinforcedCameraService
```

ShepardToTheStars commented on Resources/Prototypes/_DV/Entities/Structures/Wallmounts/WallmountMachines/reinforced_surveillance_camera.yml:
```diff
@@ -0,0 +1,279 @@
+    securityLevel: medSecurity
+  - type: Tag
+    tags:
+      - cameramed
```
```suggestion
      - ReinforcedCameraMedical
```

ShepardToTheStars commented on Resources/Prototypes/_DV/Recipes/Construction/Graphs/utilities/reinforced_cameras.yml:
```diff
@@ -0,0 +1,708 @@
+      conditions:
+      - !type:WirePanel {}
+      - !type:HasTag
+        tag: cameramed
```
```suggestion
        tag: ReinforcedCameraMedical
```

ShepardToTheStars commented on Resources/Prototypes/_DV/Recipes/Construction/Graphs/utilities/reinforced_cameras.yml:
```diff
@@ -0,0 +1,708 @@
+      conditions:
+      - !type:WirePanel {}
+      - !type:HasTag
+        tag: cameramed
```
```suggestion
        tag: ReinforcedCameraMedical
```

snowywinters committed:
Update Resources/Prototypes/_DV/Entities/Structures/Wallmounts/WallmountMachines/reinforced_surveillance_camera.yml

Co-authored-by: Vanessa <908648+ShepardToTheStars@users.noreply.github.com>
Signed-off-by: snowywinters <134970424+snowywinters@users.noreply.github.com>

snowywinters committed:
Update Resources/Prototypes/_DV/Entities/Structures/Wallmounts/WallmountMachines/reinforced_surveillance_camera.yml

Co-authored-by: Vanessa <908648+ShepardToTheStars@users.noreply.github.com>
Signed-off-by: snowywinters <134970424+snowywinters@users.noreply.github.com>

snowywinters committed:
Update Resources/Prototypes/_DV/Entities/Structures/Wallmounts/WallmountMachines/reinforced_surveillance_camera.yml

Co-authored-by: Vanessa <908648+ShepardToTheStars@users.noreply.github.com>
Signed-off-by: snowywinters <134970424+snowywinters@users.noreply.github.com>

snowywinters committed:
Update Resources/Prototypes/_DV/Entities/Structures/Wallmounts/WallmountMachines/reinforced_surveillance_camera.yml

Co-authored-by: Vanessa <908648+ShepardToTheStars@users.noreply.github.com>
Signed-off-by: snowywinters <134970424+snowywinters@users.noreply.github.com>

snowywinters committed:
Apply suggestions from code review

Co-authored-by: Vanessa <908648+ShepardToTheStars@users.noreply.github.com>
Signed-off-by: snowywinters <134970424+snowywinters@users.noreply.github.com>

snowywinters committed:
1

Co-authored-by: Vanessa <908648+ShepardToTheStars@users.noreply.github.com>
Signed-off-by: snowywinters <134970424+snowywinters@users.noreply.github.com>

snowywinters committed:
Update Resources/Prototypes/_DV/Entities/Structures/Wallmounts/WallmountMachines/reinforced_surveillance_camera.yml

Co-authored-by: Vanessa <908648+ShepardToTheStars@users.noreply.github.com>
Signed-off-by: snowywinters <134970424+snowywinters@users.noreply.github.com>

snowywinters committed:
Update Resources/Prototypes/_DV/Entities/Structures/Wallmounts/WallmountMachines/reinforced_surveillance_camera.yml

Co-authored-by: Vanessa <908648+ShepardToTheStars@users.noreply.github.com>
Signed-off-by: snowywinters <134970424+snowywinters@users.noreply.github.com>

snowywinters committed:
last part

Co-authored-by: Vanessa <908648+ShepardToTheStars@users.noreply.github.com>
Signed-off-by: snowywinters <134970424+snowywinters@users.noreply.github.com>

snowywinters committed:
now the last batch

Co-authored-by: Vanessa <908648+ShepardToTheStars@users.noreply.github.com>
Signed-off-by: snowywinters <134970424+snowywinters@users.noreply.github.com>
